.circleci/config.yml

```diff
@@ -14,11 +21,39 @@ jobs:
       - run: make image
       - run: make citest
 
+  publish:
+    machine: true
+    steps:
+      - checkout
+      - *init
+      - run:
+          name: Install Hub dependency
+          command: install_hub
+      - run:
+          name: Login on Dockerhub
+          command: login_to_dockerhub
+      - run:
+          name: Login on RubyGems
+          command: login_to_rubygems
+      - run:
+          name: Publish new version
+          command: |
+            if [ `git diff --quiet HEAD~ VERSION; echo $?` -eq 1 ]; then
```

Comment:
Can we extract this to something we can name? Trying to make it super easy to whoever is reading these steps.

Reply:
Believe me I tried. I tried creating a function for it, but it's wasn't useful it's not easy to pass the last returned code as a value. The only way is to store it in a variable but it's not an option at this step on circleci.